extensions-core/kafka-extraction-namespace/src/main/java/io/druid/query/lookup/KafkaLookupExtractorFactory.java

```diff
@@ -194,7 +195,7 @@ public boolean start()
             public void run()
             {
               while (!executorService.isShutdown()) {
-                final ConsumerConnector consumerConnector = buildConnector(kafkaProperties);
+                consumerConnector = buildConnector(kafkaProperties);
```

Comment:
Since this is in some other executor service and not guarded, there is a race condition here where `close()` can be in flight when it reaches here, missing the connector opening.

The catch here is that the `start()` method awaits this code path to make a certain level of progress, so that coordination would need to be taken account for in whatever race-condition prevention is used here.